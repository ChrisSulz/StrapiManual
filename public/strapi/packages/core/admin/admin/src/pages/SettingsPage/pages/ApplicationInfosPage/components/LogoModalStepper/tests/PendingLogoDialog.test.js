import React from 'react';

import { lightTheme, ThemeProvider } from '@strapi/design-system';
import { fireEvent, render as renderTL } from '@testing-library/react';
import { IntlProvider } from 'react-intl';

import PendingLogoDialog from '../PendingLogoDialog';

const ASSET_FIXTURES = {
  ext: '.jpeg',
  height: 250,
  name: 'asset.jpeg',
  size: 46.26,
  url: 'uploads/asset.jpeg',
  width: 340,
};

const render = (props) =>
  renderTL(
    <ThemeProvider theme={lightTheme}>
      <IntlProvider locale="en" messages={{}} textComponent="span">
        <PendingLogoDialog
          asset={ASSET_FIXTURES}
          goTo={jest.fn}
          onChangeLogo={jest.fn}
          onClose={jest.fn}
          setLocalImage={jest.fn}
          next="next"
          prev="prev"
          {...props}
        />
      </IntlProvider>
    </ThemeProvider>
  );

describe('ApplicationInfosPage | PendingLogoDialog', () => {
  it('should render and match snapshot', () => {
    const { container } = render();

    expect(container).toMatchSnapshot();
  });

  it('should call onChangeLogo and goTo callbacks', () => {
    const onChangeLogoSpy = jest.fn();
    const goToSpy = jest.fn();
    const { getByRole } = render({ onChangeLogo: onChangeLogoSpy, goTo: goToSpy });

    fireEvent.click(getByRole('button', { name: 'Upload logo' }));

    expect(onChangeLogoSpy).toHaveBeenCalledTimes(1);
    expect(goToSpy).toHaveBeenCalledTimes(1);
  });

  it('should call onChangeLogo and goTo callbacks', () => {
    const onChangeLogoSpy = jest.fn();
    const goToSpy = jest.fn();
    const { getByRole } = render({ onChangeLogo: onChangeLogoSpy, goTo: goToSpy });

    fireEvent.click(getByRole('button', { name: 'Upload logo' }));

    expect(onChangeLogoSpy).toHaveBeenCalledWith(ASSET_FIXTURES);
    expect(goToSpy).toHaveBeenCalledWith('next');
  });

  it('should call setLocalImage and goTo callbacks', () => {
    const setLocalImageSpy = jest.fn();
    const goToSpy = jest.fn();
    const { getByRole } = render({ setLocalImage: setLocalImageSpy, goTo: goToSpy });

    fireEvent.click(getByRole('button', { name: 'Choose another logo' }));

    expect(setLocalImageSpy).toHaveBeenCalledWith(undefined);
    expect(goToSpy).toHaveBeenCalledWith('prev');
  });

  it('should call onClose callback', () => {
    const onCloseSpy = jest.fn();
    const { getByRole } = render({ onClose: onCloseSpy });

    fireEvent.click(getByRole('button', { name: 'Cancel' }));

    expect(onCloseSpy).toHaveBeenCalledTimes(1);
  });
});
