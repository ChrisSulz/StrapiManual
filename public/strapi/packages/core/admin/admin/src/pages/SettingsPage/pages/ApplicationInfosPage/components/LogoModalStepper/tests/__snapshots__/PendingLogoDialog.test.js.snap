// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationInfosPage | PendingLogoDialog should render and match snapshot 1`] = `
.c33 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c5 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #32324d;
}

.c6 {
  font-size: 0.75rem;
  line-height: 1.33;
  color: #8e8ea9;
}

.c11 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  color: #ffffff;
}

.c22 {
  font-size: 0.75rem;
  line-height: 1.33;
  color: #666687;
}

.c28 {
  font-weight: 600;
  font-size: 0.6875rem;
  line-height: 1.45;
  text-transform: uppercase;
  color: #666687;
}

.c0 {
  padding-top: 24px;
  padding-right: 40px;
  padding-bottom: 24px;
  padding-left: 40px;
}

.c1 {
  padding-bottom: 24px;
}

.c7 {
  background: #4945ff;
  padding: 8px;
  padding-right: 16px;
  padding-left: 16px;
  border-radius: 4px;
  border-color: #4945ff;
  border: 1px solid #4945ff;
  cursor: pointer;
}

.c12 {
  max-width: 11.25rem;
}

.c13 {
  background: #ffffff;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  border-color: #eaeaef;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c14 {
  position: relative;
}

.c19 {
  padding-top: 8px;
  padding-right: 12px;
  padding-bottom: 8px;
  padding-left: 12px;
}

.c24 {
  background: #eaeaef;
  padding-right: 8px;
  padding-left: 8px;
  min-width: 20px;
}

.c29 {
  background: #f6f6f9;
  padding-top: 16px;
  padding-right: 20px;
  padding-bottom: 16px;
  padding-left: 20px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
}

.c15 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c20 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c25 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c26 {
  border-radius: 4px;
  height: 1.5rem;
}

.c9 {
  position: relative;
  outline: none;
}

.c9 > svg {
  height: 12px;
  width: 12px;
}

.c9 > svg > g,
.c9 > svg path {
  fill: #ffffff;
}

.c9[aria-disabled='true'] {
  pointer-events: none;
}

.c9:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c9:focus-visible {
  outline: none;
}

.c9:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c10 {
  height: 2rem;
  border: 1px solid #d9d8ff;
  background: #f0f0ff;
}

.c10 svg {
  height: 0.75rem;
  width: auto;
}

.c10[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c10[aria-disabled='true'] .c4 {
  color: #666687;
}

.c10[aria-disabled='true'] svg > g,.c10[aria-disabled='true'] svg path {
  fill: #666687;
}

.c10[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c10[aria-disabled='true']:active .c4 {
  color: #666687;
}

.c10[aria-disabled='true']:active svg > g,.c10[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c10:hover {
  background-color: #ffffff;
}

.c10:active {
  background-color: #ffffff;
  border: 1px solid #4945ff;
}

.c10:active .c4 {
  color: #4945ff;
}

.c10:active svg > g,
.c10:active svg path {
  fill: #4945ff;
}

.c10 .c4 {
  color: #271fe0;
}

.c10 svg > g,
.c10 svg path {
  fill: #271fe0;
}

.c31 {
  height: 2rem;
  border: 1px solid #dcdce4;
  background: #ffffff;
}

.c31 svg {
  height: 0.75rem;
  width: auto;
}

.c31[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c31[aria-disabled='true'] .c4 {
  color: #666687;
}

.c31[aria-disabled='true'] svg > g,.c31[aria-disabled='true'] svg path {
  fill: #666687;
}

.c31[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c31[aria-disabled='true']:active .c4 {
  color: #666687;
}

.c31[aria-disabled='true']:active svg > g,.c31[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c31:hover {
  background-color: #f6f6f9;
}

.c31:active {
  background-color: #eaeaef;
}

.c31 .c4 {
  color: #32324d;
}

.c31 svg > g,
.c31 svg path {
  fill: #32324d;
}

.c32 {
  height: 2rem;
}

.c32 svg {
  height: 0.75rem;
  width: auto;
}

.c32[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c32[aria-disabled='true'] .c4 {
  color: #666687;
}

.c32[aria-disabled='true'] svg > g,.c32[aria-disabled='true'] svg path {
  fill: #666687;
}

.c32[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c32[aria-disabled='true']:active .c4 {
  color: #666687;
}

.c32[aria-disabled='true']:active svg > g,.c32[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c32:hover {
  border: 1px solid #7b79ff;
  background: #7b79ff;
}

.c32:active {
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c32 svg > g,
.c32 svg path {
  fill: #ffffff;
}

.c18 {
  margin: 0;
  padding: 0;
  max-height: 100%;
  max-width: 100%;
  object-fit: contain;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 5.5rem;
  width: 100%;
  background: repeating-conic-gradient(#f6f6f9 0% 25%,transparent 0% 50%) 50% / 20px 20px;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
}

.c23 {
  margin-left: auto;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c27 {
  margin-left: 4px;
}

.c21 {
  word-break: break-all;
}

.c16 {
  border-bottom: 1px solid #eaeaef;
}

.c30 {
  border-radius: 0 0 4px 4px;
  border-top: 1px solid #eaeaef;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <div
        class="c3"
      >
        <span
          class="c4 c5"
        >
          Logo ready to upload
        </span>
        <span
          class="c4 c6"
        >
          Manage the chosen logo before uploading it
        </span>
      </div>
      <button
        aria-disabled="false"
        class="c7 c8 c9 c10"
        type="button"
      >
        <span
          class="c4 c11"
        >
          Choose another logo
        </span>
      </button>
    </div>
    <div
      class="c12"
    >
      <article
        aria-labelledby=":r0:-title"
        class="c13"
        tabindex="0"
      >
        <div
          class="c14 c15 c16"
        >
          <div
            class="c17"
          >
            <img
              aria-hidden="true"
              class="c18"
              src="uploads/asset.jpeg"
            />
          </div>
        </div>
        <div
          class="c19"
        >
          <div
            class="c20"
          >
            <div
              class="c21"
            >
              <div
                class="c4 c5"
                id=":r0:-title"
              >
                asset.jpeg
              </div>
              <div
                class="c4 c22"
              >
                .JPEG - 340âœ•250
              </div>
            </div>
            <div
              class="c23"
            >
              <div
                class="c24 c25 c26 c27"
              >
                <span
                  class="c4 c28"
                >
                  image
                </span>
              </div>
            </div>
          </div>
        </div>
      </article>
    </div>
  </div>
  <div
    class="c29 c30"
  >
    <div
      class="c2"
    >
      <div
        class="c8"
      >
        <button
          aria-disabled="false"
          class="c7 c8 c9 c31"
          type="button"
        >
          <span
            class="c4 c11"
          >
            Cancel
          </span>
        </button>
      </div>
      <div
        class="c8"
      >
        <button
          aria-disabled="false"
          class="c7 c8 c9 c32"
          type="button"
        >
          <span
            class="c4 c11"
          >
            Upload logo
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="c33"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
